doc/developer/platform/formalism.md

```diff
@@ -0,0 +1,176 @@
+# Materialize Formalism
+
+Materialize is a system that maintains views over changing data.
+
+At Materialize's heart are *time-varying collections* (TVCs).
+These are multisets of typed *data* whose contents may vary arbitrarily as function of some ordered *time*.
+Materialize records, transforms, and presents the contents of time varying collections.
+
+Materialize's primary function is to provide the contents of time-varying collection at specific times with absolute certainty.
+
+# Time-varying collections
+
+Materialize represents TVCs as a set of "updates" of the form `(data, time, diff)`.
+The difference `diff` represents the change in the multiplicity of `data` as we go from `time-1` to `time`.
```

Comment:
nit:  
> ... as we advance `time` to some `time' > time`.
is maybe less committing with respect to the order type (total vs partial).

Reply:
Ah, unfortunately the text is not correct in that case. Spelling out what `diff` equals for partially ordered times needs a citation to DD papers, but you are right that I should caveat this appropriately.

Reply:
Makes sense. If we should not be concerned with the partial order extensions elsewhere in this text we can probably this unchanged, citing the relevant DD papers for the curious.